onedal/neighbors/neighbors.py

```diff
@@ -597,21 +595,19 @@ def _onedal_fit(self, X, y, queue):
         return train_alg_srch.train(policy, params, to_table(X)).model
 
     def _onedal_predict(self, model, X, params, queue):
-        gpu_device = queue is not None and queue.sycl_device.is_gpu
-        if self.effective_metric_ == "euclidean" and not gpu_device:
-            if self._fit_method == "brute":
-                predict_alg = bf_knn_classification_prediction
-
-            else:
-                predict_alg = kdtree_knn_classification_prediction
+        if type(model) is kdtree_knn_classification_model:
+            return kdtree_knn_classification_prediction(**params).compute(X, model)
+        elif type(model) is bf_knn_classification_model:
+            return bf_knn_classification_prediction(**params).compute(X, model)
 
             return predict_alg(**params).compute(X, model)
 
+        gpu_device = queue is not None and queue.sycl_device.is_gpu
         policy = self._get_policy(queue, X)
         X = _convert_to_supported(policy, X)
         backend = (
             self._get_backend("neighbors", "regression", None)
-            if gpu_device
+            if queue is not None and queue.sycl_device.is_gpu
```

Comment:
please reuse `gpu_device` here.
```suggestion
            if gpu_device
```